src/OrchardCore.Modules/OrchardCore.ContentFields/Controllers/LocalizationSetContentPickerAdminController.cs

```diff
@@ -62,7 +62,7 @@ public async Task<IActionResult> SearchLocalizationSets(string part, string fiel
                 dbQuery.With<ContentItemIndex>(x => x.DisplayText.Contains(query) || x.ContentType.Contains(query));
             }
 
-            var contentItems = await dbQuery.Take(40).ListAsync();
+            var contentItems = await dbQuery.ListAsync();
```

Comment:
This can not be merged as-is as this will load all content items on every lookup. This action should accept a culture (optionally?) and then prefilter with this culture, so you only get 40 items (or whatever fixed number) of this specific culture.

Reply:
Using the localization set index is probably related to the solution.